rust/src/dcerpc/dcerpc.rs

```diff
@@ -1258,10 +1258,10 @@ pub unsafe extern "C" fn rs_dcerpc_get_tx_cnt(vtx: *mut std::os::raw::c_void) ->
 pub unsafe extern "C" fn rs_dcerpc_get_alstate_progress(tx: *mut std::os::raw::c_void, direction: u8
                                                  )-> std::os::raw::c_int {
     let tx = cast_pointer!(tx, DCERPCTransaction);
-    if direction == Direction::ToServer.into() && tx.req_done {
+    if direction == Direction::ToServer as u8 && tx.req_done {
```

Comment:
These changes do not seem needed

Could you just change `[repr(C)]` to `[repr(u8)]` ?